### PR TITLE
feature/make-worker-a-django-task

### PR DESCRIPTION
**Pull request by gecBurton**
## Context

As an Engineer I want django to control the file ingest so that:
1. we can get rid of polling the core api
2. we have better control of the file status
3. we can more reliably easily rerun file ingest 
4. we know what the ingest errors are

by using django-q2 to handle the running of the tasks we get a nice admin so that:
* we can see what tasks are in flight
* we can see which failed and why
* we can rerun bulk re-ingest etc

image

## Changes proposed in this pull request

* the worker code has been copied to `django_app/redbox/worker.py`
* django now calls the worker via `django-q2` and updates the `File.status` rather than polling core-api
* django unit tests have now been brought in line with the rest of the codebase
* `POST:core-api/file` now takes an optional param `ingest` to let it know when not to trigger the ingest
  
## Things to check

- [ ] I have added any new ENV vars in all deployed environments
- [ ] I have tested any code added or changed
- [ ] I have run integration tests https://github.com/i-dot-ai/redbox/actions/runs/10511475243

### CODE REVIEW

File: django_app/Dockerfile [hunk 1]
```diff
@@ -38,6 +39,8 @@ RUN apt-get update && apt-get install --yes libpq-dev curl > /dev/null
 
 WORKDIR /usr/src/app
 
+COPY README.md .
```
[the remark this answers is not in the document]
Comment (gecBurton) [reply django_app/Dockerfile:42]: because its mentioned in the pyproject.toml and poetry gets sad if it cant find it

File: Makefile [hunk 2]
```diff
@@ -44,14 +44,11 @@ test-ai: ## Test code with live LLM
...
-.PHONY: test-worker
-test-worker: ## Test worker
-	cd worker && poetry install && poetry run pytest --cov=worker -v --cov-report=term-missing --cov-fail-under=80
 
 .PHONY: test-django
-test-django: stop ## Test django-app
-	docker compose up -d --wait db minio
-	docker compose run --no-deps django-app venv/bin/pytest tests/ --ds redbox_app.settings -v --cov=redbox_app.redbox_core --cov-fail-under 85 -o log_cli=true
+test-django: ## Test django-app
+	cd django_app && poetry install && poetry run playwright install && poetry run pytest --cov=redbox_app -v --cov-report=term-missing --cov-fail-under=60 --ds redbox_app.settings
```
Comment (gecBurton) [inline Makefile:50]:
```suggestion
	cd django_app && poetry install && poetry run playwright install && poetry run pytest tests/ --cov=redbox_app -v --cov-report=term-missing --cov-fail-under=60 --ds redbox_app.settings
```

File: redbox-core/redbox/chains/ingest.py [hunk 3]
```diff
@@ -43,5 +43,5 @@ def ingest_from_loader(
         document_loader(document_loader_type=document_loader_type, s3_client=s3_client, env=env)
         | RunnableLambda(list)
         | log_chunks
-        | RunnableLambda(partial(vectorstore.aadd_documents, create_index_if_not_exists=False))  # type: ignore[arg-type]
```
Comment (jamesrichards4) [inline redbox-core/redbox/chains/ingest.py:46]: can we leave this as the async version? I don't think it matters hugely but if we can async we should

File: redbox-core/redbox/models/chat.py [hunk 4]
```diff
@@ -22,7 +22,7 @@ class ChatMessage(BaseModel):
 
 
 class SelectedDocument(BaseModel):
-    uuid: UUID | None = Field(description="uuid of selected file", default=None)
```
[the remark this answers is not in the document]
Comment (gecBurton) [reply redbox-core/redbox/models/chat.py:25]: good point, i forgot to remove it

File: Makefile [hunk 5]
```diff
@@ -44,14 +44,11 @@ test-ai: ## Test code with live LLM
...
 
-.PHONY: test-worker
-test-worker: ## Test worker
-	cd worker && poetry install && poetry run pytest --cov=worker -v --cov-report=term-missing --cov-fail-under=80
 
 .PHONY: test-django
-test-django: stop ## Test django-app
-	docker compose up -d --wait db minio
-	docker compose run --no-deps django-app venv/bin/pytest tests/ --ds redbox_app.settings -v --cov=redbox_app.redbox_core --cov-fail-under 85 -o log_cli=true
+test-django: ## Test django-app
```
[the remark this answers is not in the document]
Comment (gecBurton) [reply Makefile:49]: i was hoping you wouldnt notice this!

File: core-api/core_api/routes/file.py [hunk 6]
```diff
@@ -76,7 +76,9 @@ class FileRequest(BaseModel):
```
[the remark this answers is not in the document]
Comment (gecBurton) [reply core-api/core_api/routes/file.py]: now gone

**Pull request merged**